src/OrchardCore.Themes/TheAdmin/Views/Navigation-admin.cshtml

```diff
@@ -1,14 +1,56 @@
 @{
     TagBuilder tag = Tag(Model, "ul");
-
+    tag.Attributes["Id"] = "adminMenu";
 
     foreach (var item in Model)
     {
         tag.InnerHtml.AppendHtml(await DisplayAsync(item));
     }
 }
 
+<input id="filter" class="form-control" placeholder="@T["Filter"]" type="search" data-toggle="tooltip" data-html="true" title="Ctrl+Shift+F" />
 <nav id="left-nav">
     @tag
 </nav>
-<!-- Menu rendered on @DateTime.UtcNow.ToString("r") -->
\ No newline at end of file
+<script at="Foot">        
+    $('[data-toggle="tooltip"]').tooltip();
+    $(document).bind('keydown', function(e) {
+        if (e.ctrlKey && e.shiftKey && e.which == 70) {
+            $('#filter').focus();
+            return false;
+        }
+    });
+    // custom css expression for a case-insensitive contains()
+    jQuery.expr[":"].contains = jQuery.expr.createPseudo(function (arg) {
+        return function (elem) {
+            return jQuery(elem).text().toUpperCase().indexOf(arg.toUpperCase()) >= 0;
+        };
+    });
+
+    function hasChild(list) {
+        var filter = $('#filter').val();
+        list.children('li').each(function () {
+            if ($(this).find("span:contains(" + filter + ")").length > 0) {
+                if ($(this).find('ul :first').length > 0) {
+                    $(this).show();
+                    hasChild($(this).find('ul :first').parent());
+                } else {
+                    $(this).show();
+                }
+            } else {
+                $(this).hide();
+            }
+        });
+    };
+    $('#filter').keyup(function () {
+        var list = '#adminMenu';
```

Comment:
maybe you meant to initalize a jquery object instead of a string. No need to reuse the string.